sympy/functions/elementary/exponential.py

```diff
@@ -1234,30 +1234,54 @@ def _eval_is_algebraic(self):
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         if len(self.args) == 1:
             arg = self.args[0]
-            arg0 = arg.subs(x, 0).cancel()
-            if not arg0.is_zero:
-                return self.func(arg0)
-            return arg.as_leading_term(x)
+            _, e = arg.as_coeff_exponent(x)
+            if e.is_nonnegative:
+                arg0 = arg.subs(x, 0).cancel()
+                if not arg0.is_zero:
+                    return self.func(arg0)
+                return arg.as_leading_term(x)
 
     def _eval_nseries(self, x, n, logx, cdir=0):
         if len(self.args) == 1:
             from sympy.functions.elementary.integers import ceiling
             from sympy.series.order import Order
             arg = self.args[0].nseries(x, n=n, logx=logx)
             lt = arg.as_leading_term(x, logx=logx)
-            lte = 1
+            lte = S.One
             if lt.is_Pow:
                 lte = lt.exp
-            if ceiling(n/lte) >= 1:
-                s = Add(*[(-S.One)**(k - 1)*Integer(k)**(k - 2)/
-                          factorial(k - 1)*arg**k for k in range(1, ceiling(n/lte))])
-                s = expand_multinomial(s)
-            else:
-                s = S.Zero
-
-            return s + Order(x**n, x)
+            if lte.is_positive:
+                if (ceiling(n / lte) >= 1) is S.true:
+                    s = Add(*[(-S.One)**(k - 1)*Integer(k)**(k - 2)/
```

Comment:
I think `S.NegativeOne` is more appropriate since it is provided, rather than using `-S.One`.